rust/operator-binary/src/main.rs

```diff
@@ -148,9 +149,16 @@ async fn main() -> anyhow::Result<()> {
 }
 
 fn references_authentication_class(
-    trino: &TrinoCluster,
-    authentication_class: &AuthenticationClass,
+    trino: &DeserializeGuard<TrinoCluster>,
+    authentication_class: &DeserializeGuard<AuthenticationClass>,
 ) -> bool {
+    let Ok(trino) = &trino.0 else {
+        return false;
+    };
+    let Ok(authentication_class) = &authentication_class.0 else {
+        return false;
+    };
```

Comment:
No need for this, `DeserializeGuard` still provides meta access.

Reply:
Good to know. [Thanks](https://github.com/stackabletech/trino-operator/pull/657/commits/63e0a4fa048a2311ef98197873bcd2d283d12bda)